src/modules/fundingManager/extensions/FM_EXT_TokenVault_v1.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+// Internal Dependencies
+import {Module_v1} from "src/modules/base/Module_v1.sol";
+import {IFM_EXT_TokenVault_v1} from
+    "src/modules/fundingManager/extensions/interfaces/IFM_EXT_TokenVault_v1.sol";
+
+// External Dependencies
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {ERC165Upgradeable} from
+    "@oz-up/utils/introspection/ERC165Upgradeable.sol";
+
+/**
+ * @title   Token Vault
+ *
+ * @notice  Pool holding token Reserves for later use.
+ *
+ * @dev     Funds can be withdrawn by OrchestratorAdmin
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to our Security Policy
+ *                          at security.inverter.network or email us directly!
+ *
+ * @author  Inverter Network
+ */
+contract FM_EXT_TokenVault_v1 is IFM_EXT_TokenVault_v1, Module_v1 {
```

Comment:
Because its an extension of a FundingManager
We did the same for the Authorizer Extension
Also for it being a logicmodule it would have to implement the ERC20PaymentClient, which it also doesnt do